code/modules/cargo/centcom_podlauncher.dm

```diff
@@ -39,48 +44,118 @@
 	var/list/turf/acceptableTurfs = list() //Contians a list of turfs (in the "bay" area on centcom) that have items that can be launched. Taken from orderedArea
 	var/list/launchList = list() //Contains whatever is going to be put in the supplypod and fired. Taken from acceptableTurfs
-	var/obj/effect/supplypod_selector/selector = new() //An effect used for keeping track of what item is going to be launched when in "ordered" mode (launchChoice = 1)
+	var/obj/effect/hallucination/simple/supplypod_selector/selector //An effect used for keeping track of what item is going to be launched when in "ordered" mode (launchChoice = 1)
+	var/obj/effect/hallucination/simple/dropoff_location/indicator
 	var/obj/structure/closet/supplypod/centcompod/temp_pod //The temporary pod that is modified by this datum, then cloned. The buildObject() clone of this pod is what is launched
+	// Stuff needed to render the map
+	var/map_name
+	var/obj/screen/map_view/cam_screen
+	var/list/cam_plane_masters
+	var/obj/screen/background/cam_background
+	var/tabIndex = 1
+	var/list/timers = list("landingDelay", "fallDuration", "openingDelay", "departureDelay")
+	var/renderLighting = FALSE
 
-/datum/centcom_podlauncher/New(H)//H can either be a client or a mob due to byondcode(tm)
-	if (istype(H,/client))
-		var/client/C = H
-		holder = C //if its a client, assign it to holder
+/datum/centcom_podlauncher/New(user) //user can either be a client or a mob
+	if (user) //Prevents runtimes on datums being made without clients
+		setup(user)
+
+/datum/centcom_podlauncher/proc/setup(user) //H can either be a client or a mob
+	if (istype(user,/client))
+		var/client/user_client = user
+		holder = user_client //if its a client, assign it to holder
 	else
-		var/mob/M = H
-		holder = M.client //if its a mob, assign the mob's client to holder
+		var/mob/user_mob = user
+		holder = user_mob.client //if its a mob, assign the mob's client to holder
 	bay =  locate(/area/centcom/supplypod/loading/one) in GLOB.sortedAreas //Locate the default bay (one) from the centcom map
-	temp_pod = new(locate(/area/centcom/supplypod/pod_storage) in GLOB.sortedAreas) //Create a new temp_pod in the podStorage area on centcom (so users are free to look at it and change other variables if needed)
+	var/list/pick_turfs = list()
+	var/area/pod_storage_area = locate(/area/centcom/supplypod/pod_storage) in GLOB.sortedAreas
+	for(var/turf/pod_storage_turf in pod_storage_area)
+		pick_turfs += pod_storage_turf
+	temp_pod = new(pick(pick_turfs)) //Create a new temp_pod in the podStorage area on centcom (so users are free to look at it and change other variables if needed)
 	orderedArea = createOrderedArea(bay) //Order all the turfs in the selected bay (top left to bottom right) to a single list. Used for the "ordered" mode (launchChoice = 1)
+	var/turf/picked_bay_turf = pick(bay)
```

Comment:
While this does work, I believe I intended for it to be pick(get_area_turfs(bay)), etc. 

Instead of choosing a random turf, this here always picks the south-westernmost turf. Weird behavior that has honestly worked out fine for my use cases. I'll switch it over to use get_area_turfs() as I feel like there's some byond fuckery going on with this